shared/constants/types/fs.js

```diff
@@ -3,23 +3,35 @@ import * as I from 'immutable'
 
 export opaque type Path = ?string
 
-export type PathType = 'folder' | 'file' | 'symlink' | 'exec' | 'unknown'
+export type PathType = 'folder' | 'file' | 'symlink' | 'unknown'
+
+type pendingPathMetadataProgressType = 1
+type loadedPathMetadataProgressType = 2
+export const PendingPathMetadataProgress = 1
+export const LoadedPathMetadataProgress = 2
+
+export type PathItemMetadata = {
+  lastModifiedTimestamp?: number,
```

Comment:
might be better to avoid nullable if possible. just makes you type less code later.
could say if timestamp is 0 it means nothings been written and its always non null

Reply:
I tried to make this an intersection type based on the value of `progress`, but I couldn't get Flow to type things correctly with that.